core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoBigQuerySupport.scala

```diff
@@ -133,6 +140,86 @@ trait DataRepoBigQuerySupport {
     }
   }
 
+  /**
+   * Translates a BigQuery result set into the pagination metadata that Rawls expects.
+   * @param queryResults the BigQuery result set
+   * @param entityQuery the query criteria supplied by the user, which includes page size
+   * @return the Rawls-flavor pagination metadata
+   */
+  def queryResultsMetadata(queryResults:TableResult, entityQuery: EntityQuery): EntityQueryResultMetadata = {
+    val totalRowCount = queryResults.getTotalRows
+
+    val pageCount = Math.ceil(totalRowCount.toFloat / entityQuery.pageSize).toInt
+
+    // we don't support filtering in BQ, so unfilteredCount and filteredCount are the same
+    EntityQueryResultMetadata(totalRowCount.toInt, totalRowCount.toInt, pageCount)
+  }
+
+  /**
+   * Given an EntityQuery, which contains page number and page size, generate the correct absolute pagination offset for use in BQ
+   * @param entityQuery the query criteria supplied by the user
+   * @return the offset value to use in an OFFSET sql clause
+   */
+  def translatePaginationOffset(entityQuery: EntityQuery): Int = {
+    if (entityQuery.page < 1)
+      throw new DataEntityException("page value must be at least 1.", code = StatusCodes.BadRequest)
+    (entityQuery.page-1) * entityQuery.pageSize
+  }
+
+  /**
+   * generates the SQL and config to send to BigQuery for use in the queryEntities() method
+   * @param dataProject project containing the BQ data
+   * @param viewName dataset/snapshot to query in BQ
+   * @param entityType table to query in BQ
+   * @param entityQuery user-supplied query criteria
+   * @return the object to pass to BQ to execute a query
+   */
+  def queryConfigForQueryEntities(dataProject: String, viewName: String, entityType: String, entityQuery: EntityQuery): QueryJobConfiguration = {
+    // generate BQ SQL for this entity
+    val query = s"SELECT * FROM `${sanitizeSql(dataProject)}.${sanitizeSql(viewName)}.${sanitizeSql(entityType)}` " +
```

Comment:
why do we sanitize dataProject, viewName and entityType (but we must verify that entityType exists in snapshot) since those come from data repo? For entityQuery.sortField we could verify that the field exists in the snapshot schema.

Reply:
I do both just for layered security. I don't think doing both HURTS - beyond the need to maintain the additional code. Do you think it's not worth it? FYI, I do already check that the sort field exists in the schema.

Reply:
I support being extra cautious.